firmware/main.c

```diff
@@ -692,131 +689,128 @@ static void cmd_motor_rotary(BaseSequentialStream *chp, int argc, char *argv[])
-			"\t  0 - CW\r\n"
-			"\t  1 - CCW\r\n");
+			"\t 'home' - move to reference position to calibrate angle measurement\r\n"
+			"\t 'rotate' <pulses>\r\n"
+			"\t\t <pulses> Option 1: +(number of pulses as int) - clockwise (CW) rotation\r\n"
+			"\t\t <pulses> Option 2: -(number of pulses as int) - counterclockwise (CCW) rotation\r\n"
+			);
 }
 
 
 static void cmd_motor_linear(BaseSequentialStream *chp, int argc, char *argv[]) {
+	// Pin declaration for use with the C5-E (gpio number arbitrary)
 	const char maplePinPulses[] = "26";
 	const char maplePinDirection[] = "28";
 	const char maplePinEnable[] = "22";
 
 	int pinPulses, pinDirection, pinEnable;
 
-	char * paramMode = NULL;
-	int mode = -1;
+	// available modes
+	const int MOVE = 1, HOME = 2;
 
-	char * paramDirection = NULL;
+	int mode = -1;
 	int direction = -1;
-
-	char * paramPulses = NULL;
 	int pulses = -1;
 
+	// define motion directions: towards switchboard/wall
+	int SWITCHBOARD = 0, WALL = 1;
+
 	int i;
 
 	// Parsing
 	for(i = 0; i < argc; i++) {
-		if(strcmp(argv[i], "-d") == 0) {
+		if(strcmp(argv[i], "move") == 0) {
 			if(++i >= argc)
 				continue;
-			paramDirection = argv[i];
-		}
-		else if(strcmp(argv[i], "-p") == 0) {
-			if(++i >= argc)
-				continue;
-			paramPulses = argv[i];
+			pulses = atoi(argv[i]);
+			mode = MOVE;
+			// direction is set by presign of the given argument
+			if(pulses >= 0)
+				direction = SWITCHBOARD;
+			else if (pulses < 0) {
+				direction = WALL;
+				// Direction set by direction parameter
+				// pulses need to be a positive int
+				pulses = abs(pulses);
+			}
 		}
-		else if(strcmp(argv[i], "-m") == 0) {
+		else if(strcmp(argv[i], "home") == 0) {
+			mode = HOME;
 			if(++i >= argc)
 				continue;
-			paramMode = argv[i];
 		}
 	}
 
-	if(!paramMode)
-		goto exit_with_usage;
-	mode = atoi(paramMode);
-	if((mode < 1) || (mode > 2))
+	if(!(mode == MOVE || mode == HOME))
 		goto exit_with_usage;
 
-	if(1 == mode)
-		if(!paramPulses || !paramDirection)
-			goto exit_with_usage;
-	if(2 == mode)
-		if(paramPulses || paramDirection)
-			goto exit_with_usage;
-
-	if(paramDirection) {
-		direction = atoi(paramDirection);
-		if((direction < 0) || (direction > 1))
-			goto exit_with_usage;
-	}
-
-	if(paramPulses)
-		pulses = atoi(paramPulses);
-
 	/* Pin definition for the pulse and direction outputs */
 
-	if(1 == mode) {
-		// Configure the pins just when mode 1 is selected
+	if(mode == MOVE) {
+		// Configure the pins just when the mode "move" is used and pulses will be send
 		pinPulses = pinIndexForMaplePin(maplePinPulses, true, false, true);
 		palSetPadMode(pinPorts[pinPulses].gpio, pinPorts[pinPulses].pin, PAL_MODE_OUTPUT_PUSHPULL);
 		palClearPad(pinPorts[pinPulses].gpio, pinPorts[pinPulses].pin);
 
 		pinDirection = pinIndexForMaplePin(maplePinDirection, true, false, true);
 		palSetPadMode(pinPorts[pinDirection].gpio, pinPorts[pinDirection].pin, PAL_MODE_OUTPUT_PUSHPULL);
-		if(1 == direction)
+		if(direction == SWITCHBOARD)
```

Comment:
i know i was not very consistent about this, but in equality comparisons i always prefer yoda conditions.